src/cranbtree.c

```diff
@@ -256,8 +259,11 @@ void *cbt_search(cranbtree_t * bt, int key)
   */
 void *cbt_delete(cranbtree_t * bt, int key)
 {
-	assert(bt != NULL);
-
+	//assert(bt != NULL);
+	if (bt == NULL)
+	{
+		bt->op_errno = CBT_KEY_NOT_FOUND;
```

Comment:
This line would cause a segmentation fault because at this point you would be dereferencing a NULL pointer.